src/service.rs

```diff
@@ -1026,8 +1022,14 @@ impl Service {
     ) {
         // find the ENR associated with the query
         if let Some(enr) = self.find_enr(&return_peer) {
+            let contact = match NodeContact::try_from_enr(enr) {
+                Ok(contact) => contact,
+                Err(NonContactable) => {
+                    return error!("Query {} has a non contactable enr", *query_id)
```

Comment:
I think we have to be careful here. 

A query is a larger group of RPC requests. I think in the previous version, we would send the request to the handler, the handler would find the ENR to be non-contactable, return a RequestError then we would mark the query request as failed and the query would be able to progress (its a big state machine)

If we are circumventing the handler here, I think we need to tell the query that this RPC request failed, so it can progress. 

This logic is complex and convoluted. Perhaps we should design it in a way that we can identify these kinds of things at compile time

Reply:
I'm leaving this one to you since I haven't studied this part of the code yet and I don't find it feasible to study it soon / fast enough